### DISCUSSION TITLE
open menu when mouseenter on button

### DISCUSSION THREAD

**Discussion by TeeFiX42** [zauberzeug/nicegui#3091] [Q&A] [answered]
Hi,
I have a group of buttons that open each a different menu when the user clicks on the button.
I want to open the menu (of that button) when the mouse enters the button ("mouseenter") event.
```py
for group in menu_data.groups:
     with ui.button(text=group.group_display_name) as group_button:  
         with ui.menu().props("anchor='top end'") as group_menu:
             for submenu in group.submenus:
```

this is small snippet of how the list of buttons and menus is generated.
How can I make the menu visible when the mouse enters the button?

**Comment by falkoschindler** [comment]
Hi @TeeFiX42,

Do you mean something like this?
```py
with ui.button('Button').on('mouseenter', lambda: group_menu.open()):
    with ui.menu().props("anchor='top end'") as group_menu:
        ui.menu_item('Item')
```

**Reply by TeeFiX42** [reply]
Yes, but with this method, the group_menu of the last group_button is opened and not the menu of the button that was entered.

**Reply by falkoschindler** [reply, marked as answer]
That's a common problem with Python's "late binding":
https://github.com/zauberzeug/nicegui/wiki/FAQs#why-is-the-last-element-of-my-for-loop-used-in-every-step

**Reply by TeeFiX42** [reply]
Tnx!
Tnx for answering so quick.
Tnx for the framework!
